stage0/src/image_header.rs

```diff
@@ -2,124 +2,34 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+use abi::ImageHeader;
+
 extern "C" {
-    static IMAGEA: ImageHeader;
+    static IMAGEA: abi::ImageHeader;
 }
 
-// TODO grab this from lpc55_support or another crate eventually
-#[derive(Debug)]
-#[repr(C)]
-struct CertHeader {
-    signature: [u8; 4],
-    header_version: u32,
-    header_length: u32,
-    flags: u32,
-    build_number: u32,
-    total_image_len: u32,
-    certificate_count: u32,
-    certificate_table_len: u32,
-    key_size: u32,
-    // The u32 here represents the start of the key which comes after this
-    // structure.
-    key: u32,
-}
+pub struct Image(&'static ImageHeader);
 
-pub fn get_image_a() -> Option<&'static ImageHeader> {
+pub fn get_image_a() -> Option<Image> {
     // Taking the reference to our supposed imagea
```

Comment:
Here's a better safety comment:

```rust
// Safety: this is unsafe because `IMAGEA` is coming from
// an extern, and might violate alignment rules or even be
// modified externally and subject to data races. In our case
// we have to assume that neither of these is true, since it's
// being furnished by our linker script, which we trust.
```